tslib/react/src/components/TrialTable.tsx

```diff
@@ -1,153 +1,103 @@
 import { FC } from "react"
 
 import * as Optuna from "@optuna/types"
-import { DataGrid, DataGridColumn } from "./DataGrid"
+import { DataGrid } from "./DataGrid"
+
+import {
+  ColumnDef,
+  FilterFn,
+  Row,
+  createColumnHelper,
+} from "@tanstack/react-table"
+
+const multiValueFilter: FilterFn<Optuna.Trial> = <D extends object>(
+  row: Row<D>,
+  columnId: string,
+  filterValue: string[]
+) => {
+  const rowValue = row.getValue(columnId) as string
+  return !filterValue.includes(rowValue)
+}
 
 export const TrialTable: FC<{
   study: Optuna.Study
-  initialRowsPerPage?: number
-}> = ({ study, initialRowsPerPage }) => {
+}> = ({ study }) => {
```

Comment:
Following the comment above, maybe you need to restore `initialRowsPerPage` prop here as well.
It would be also good in terms of not having to make destructive changes to the existing interface.